src/services/api.ts

```diff
@@ -25,13 +26,13 @@ import {
 axios.defaults.baseURL = API_URL;
 
 export class APIService {
-  provider = undefined;
+  provider = undefined as Web3Provider | undefined;
```

Comment:
is there a way to type this just as `Web3Provider`? we'd never want it to be `undefined`, right?

Reply:
I think you need to be connected with a wallet to have a provider